### PR TITLE
test(dtslint): add dtslint test for pluck operator(#4093)

### PR DESCRIPTION
**Pull request by dkosasih**
**Description:** Add dtslint test for Pluck operator

**Related issue (if exists):** #4093

### CODE REVIEW

File: spec-dtslint/operators/pluck-spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+it('should infer generic type', () => {
+  const a = of([{ name: 'abc', id: 123 }, { name: 'def', id: 256 }]).pipe<string>(pluck('name')); // $ExpectType Observable<string>
+});
+
+it('should support nested object', () => {
+  const a = of([{ name: 'abc', id: 123, address: { postcode: 1 } }, { name: 'def', id: 256, address: { postcode: 2 } }]).pipe(pluck('address', 'postcode')); // $ExpectType Observable<{}
+});
+
+it('should accept string only', () => {
+  const a = of([{ name: 'abc', id: 123}, { name: 'def', id: 256}]).pipe(pluck(1)); // $ExpectError
```
Comment (timdeschryver) [inline spec-dtslint/operators/pluck-spec.ts]: Should we also add a test that enforces a parameter?

```ts
const a = of([{ name: 'abc', id: 123}, { name: 'def', id: 256}]).pipe(pluck()); // $ExpectError
```

**Comment by dkosasih** [comment]
Should I port that over?
I don't mind doing a PR for that.

**Comment by cartant** [comment]
@dkosasih Sure, if you want. There is an issue here: #4131.

It will also need a rest parameters overload like the one in `pipe` - i.e. with trailing rest parameters.

**Comment by cartant** [comment]
Actually, using a trailing rest parameters signature would be a breaking change. The typed signatures should be added and its current signature should be kept.

**Comment by dkosasih** [comment]
@cartant I think I lost you there :smile:

**Comment by dkosasih** [comment]
I have change the tests for this operator - this should only be be merge if the pull request for #4131 is approved.
Thanks.

File: spec-dtslint/operators/pluck-spec.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+              aa: 1
+            }
+          }
+        }
+      }
+    }
+  }).pipe(pluck('address', 'postcode', 'x', 'y', 'z', 'aa')); // $ExpectType Observable<number>
+});
+
+it('should support nested object of 6 layer depth', () => {
```
Comment (cartant) [inline spec-dtslint/operators/pluck-spec.ts]: Isn't this the same as the test above?

File: spec-dtslint/operators/pluck-spec.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          }
+        }
+      }
+    }
+  }).pipe(pluck('address', 'postcode', 'x', 'y', 'z', 'aa', 'ab')); // $ExpectType Observable<{}>
+});
+
+it('should accept string only', () => {
+  const a = of({ name: 'abc', id: 123}).pipe(pluck(1)); // $ExpectError
+});
```
Comment (cartant) [inline spec-dtslint/operators/pluck-spec.ts]: Another test that shows that if a string that is not a key of the type is specified,  `{}` should be inferred, as it should fall through to the rest parameters signature.

**Comment by dkosasih** [comment]
@cartant Do you think we could merge this test already since the PR for #4131 has been merged?

**Comment by cartant** [comment]
You will need to rebase this PR against `master`, first, so that the tests pass.

**Pull request merged**